client/extensions/woocommerce/app/settings/email/mailchimp/setup-mailchimp.js

```diff
@@ -0,0 +1,315 @@
+			return;
+		} else if ( this.state.step === CAMPAIGN_DEFAULTS_STEP ) {
+			const settings = this.getCampaingDefaultsSettings();
+			const validSettings = this.areCampaignSettingsValid( settings );
+			if ( ! validSettings ) {
+				this.setState( { settings_values_missing: true } );
+				return;
+			}
+			this.setState( { settings_values_missing: false } );
+			this.props.submitMailchimpCampaignDefaults( this.props.siteId, settings );
+			return;
+		} else if ( this.state.step === NEWSLETTER_SETTINGS_STEP ) {
+			const mailchimp_list = this.state.settings.mailchimp_list;
+			if ( ! mailchimp_list ) {
+				this.setState( { settings_values_missing: true } );
+				return;
+			}
+			this.setState( { settings_values_missing: false } );
+			this.props.submitMailchimpNewsletterSettings( this.props.siteId, { mailchimp_list } );
+			return;
+		}
+		this.setState( { step: steps[ this.state.step ].nextStep } );
+	}
+
+	onKeyInputChange = ( e ) => {
+		this.setState( { api_key_input: e.target.value } );
+	}
+
+	// Right now Store info is combination of values from SettingsPaymentsLocationCurrency
+	// and managed directly - not the greatest option but good for now.
+	onStoreInfoChange = ( e ) => {
+		this.setState( { settings: Object.assign( {}, this.state.settings, { [ e.target.name ]: e.target.value } ) } );
+	}
+
+	renderStep = () => {
+		const { step } = this.state;
+		if ( step === LOG_INTO_MAILCHIMP_STEP ) {
+			return <LogIntoMailchimp />;
+		}
+		if ( step === KEY_INPUT_STEP ) {
+			return <KeyInputStep
+				onChange={ this.onKeyInputChange }
+				apiKey={ this.state.api_key_input }
+				isKeyCorrect={ this.props.isKeyCorrect } />;
+		}
+		if ( step === STORE_INFO_STEP ) {
+			return <StoreInfoStep
+				onChange={ this.onStoreInfoChange }
+				storeData={ this.state.settings }
+				validateFields={ false }
+			/>;
+		}
+		if ( step === CAMPAIGN_DEFAULTS_STEP ) {
+			return <CampaignDefaultsStep
+				onChange={ this.onStoreInfoChange }
+				storeData={ this.state.settings }
+				validateFields={ false }
+			/>;
+		}
+		if ( step === NEWSLETTER_SETTINGS_STEP ) {
+			return <NewsletterSettings
+				onChange={ this.onStoreInfoChange }
+				storeData={ this.state.settings }
+				validateFields={ false }
+			/>;
+		}
+		return <div></div>;
+	}
+
+	render() {
+		const { translate, hasMailChimp, siteId } = this.props;
+		const isButtonBusy = this.props.isBusy ? 'is-busy' : '';
+		const buttons = [
+			{ action: 'cancel', label: translate( 'Cancel' ) },
+			{ action: 'next', label: translate( 'Next' ), onClick: this.next, isPrimary: true, additionalClassNames: isButtonBusy },
+		];
+
+		const dialogClass = 'woocommerce mailchimp__setup';
+		const stepNum = steps[ this.state.step ].number;
+		if ( ! hasMailChimp ) {
+			return (
+				<Dialog
+					isVisible={ true }
+					buttons={ null }
+					className={ dialogClass }>
+					<div className="mailchimp__setup-dialog-title">MailChimp</div>
+					<RequiredPluginsInstallView
+						site={ { ID: siteId } }
```

Comment:
since the only item being passed here is the `site.Id` wondering if `RequiredPluginsInstallView` should be refactored a bit. Definitely could be done in a follow-up PR.

Reply:
Good idea. Let's keep interfaces simple. Follow-up PR.